cellprofiler_core/readers/bioformats_reader.py

```diff
@@ -65,18 +79,140 @@ def read(self,
         :param wants_max_intensity: if `False`, only return the image; if `True`,
                   return a tuple of image and max intensity
         :param channel_names: provide the channel names for the OME metadata
+        :param XYWH: a (x, y, w, h) tuple
         """
-        reader = self.get_reader()
-        return reader.read(
-            c=c,
-            z=z,
-            t=t,
-            series=series,
-            index=index,
-            rescale=rescale,
-            wants_max_intensity=wants_max_intensity,
-            channel_names=channel_names,
-        )
+        self._ensure_file_open()
+
+        FormatTools = jimport("loci.formats.FormatTools")
+        ChannelSeparator = jimport("loci.formats.ChannelSeparator")
+        if series is not None:
+            self._reader.setSeries(series)
+
+        if XYWH is not None:
+            assert isinstance(XYWH, tuple) and len(XYWH) == 4, "Invalid XYWH tuple"
+            openBytes_func = lambda x: self._reader.openBytes(x, XYWH[0], XYWH[1], XYWH[2], XYWH[3])
+            width, height = XYWH[2], XYWH[3]
+        else:
+            openBytes_func = self._reader.openBytes
+            width, height = self._reader.getSizeX(), self._reader.getSizeY()
+
+        # FIXME instead of np.frombuffer use scyjava.to_python, ideally that wraps memory
```

Comment:
It seems like we now have a mix of the two, is this note intended to resolve the later ones (and if so, is there a reason not to just do it now, or is the one place where it's now used meant to be the only place and thus can it be resolved?